File: frontend/src/api/vocabulary.ts
import { apiRequest } from './apiClient'

export type VocabularyItem = {
  id: string
  word: string
  translation: string
  context_sentence: string
  mastery_level: number
  next_review_at: string
}

export type VocabularyReviewRequest = {
  vocabulary_id: string
  rating: number
}

export const vocabularyApi = {
  daily: () => apiRequest<VocabularyItem[]>({ path: '/vocabulary/daily-review', method: 'GET' }),
  review: (body: VocabularyReviewRequest) => apiRequest<any>({ path: '/vocabulary/review', method: 'POST', body }),
}

File: backend/app/schemas/vocabulary.py
from pydantic import BaseModel
from uuid import UUID
from datetime import datetime

class VocabularyItemResponse(BaseModel):
    word: str
    translation: str
    context_sentence: str
    id: UUID
    mastery_level: int
    next_review_at: datetime

    class Config:
        from_attributes = True

class VocabularyReviewRequest(BaseModel):
    vocabulary_id: UUID
    rating: int # 1-5
